bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,356 @@
+    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
+    memcpy(blobs[blob_index], blob_bytes, BYTES_PER_BLOB);
+  }
+
+  KZGProof proof;
+  C_KZG_RET ret = compute_aggregate_kzg_proof(
+    &proof,
+    blobs,
+    blobs_count,
+    kzg_settings
+  );
+
+  if (ret != C_KZG_OK) {
+     Napi::Error::New(env, "Failed to compute proof")
+      .ThrowAsJavaScriptException();
+
+    free(blobs);
+    return env.Undefined();
+  };
+
+  free(blobs);
+  uint8_t proof_bytes[BYTES_PER_PROOF];
+  bytes_from_g1(proof_bytes, &proof);
+  return napi_typed_array_from_bytes(proof_bytes, BYTES_PER_PROOF, env);
+}
+
+// verifyAggregateKzgProof: (blobs: Blob[], expectedKzgCommitments: KZGCommitment[], kzgAggregatedProof: KZGProof, setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyAggregateKzgProof(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 4;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto blobs_param = info[0].As<Napi::Array>();
+  auto comittments_param = info[1].As<Napi::Array>();
+  auto proof_param = info[2].As<Napi::TypedArray>();
+  auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
+
+  auto proof_bytes = proof_param.As<Napi::Uint8Array>().Data();
+  auto blobs_count = blobs_param.Length();
+
+  auto blobs = (Blob*)calloc(blobs_count, sizeof(Blob));
+  auto commitments = (KZGCommitment*)calloc(blobs_count, sizeof(KZGCommitment));
+
+  C_KZG_RET ret;
+
+  for (uint32_t blob_index = 0; blob_index < blobs_count; blob_index++) {
+    // Extract blob bytes from parameter
+    Napi::Value blob = blobs_param[blob_index];
+    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
+
+    memcpy(blobs[blob_index], blob_bytes, BYTES_PER_BLOB);
+
+    // Extract a G1 point for each commitment
+    Napi::Value commitment = comittments_param[blob_index];
+    auto commitment_bytes = commitment.As<Napi::Uint8Array>().Data();
+
+    ret = bytes_to_g1(&commitments[blob_index], commitment_bytes);
+    if (ret != C_KZG_OK) {
+      std::ostringstream ss;
+      std::copy(commitment_bytes, commitment_bytes + BYTES_PER_COMMITMENT, std::ostream_iterator<int>(ss, ","));
+
+      Napi::TypeError::New(
+        env,
+        "Invalid commitment data"
+      ).ThrowAsJavaScriptException();
+
+      free(commitments);
+      free(blobs);
+
+      return env.Null();
+    }
+  }
+
+  KZGProof proof;
+  ret = bytes_to_g1(&proof, proof_bytes);
+  if (ret != C_KZG_OK) {
+    free(commitments);
+    free(blobs);
+
+    Napi::Error::New(env, "Invalid proof data")
+      .ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  bool verification_result;
+  ret = verify_aggregate_kzg_proof(
+    &verification_result,
+    blobs,
+    commitments,
+    blobs_count,
+    &proof,
+    kzg_settings
+  );
+  if (ret != C_KZG_OK) {
+    free(commitments);
```

Comment:
could do these `free`s above the `if` (and remove the copy after the `if`) -- your current way may be more consistent in style though.